algebra/src/biginteger/mod.rs

```diff
@@ -21,11 +16,52 @@ bigint_impl!(BigInteger384, 6);
 bigint_impl!(BigInteger768, 12);
 bigint_impl!(BigInteger832, 13);
 
+impl<T: BigInteger> CanonicalSerialize for T {
```

Comment:
Does this replace the Fp serialization?
If so, there's a significant disadvantage to just relying on the limbs for the serialized size. BigIntegers are often longer than the requires bytes needed to encode the field elements. Since big integers sizes are multiples of 64 and the field size is a multiple of 8, there would be always-0 bytes in the last limb, which affects the serialized size significantly.

Reply:
Did the old implementation avoid this pitfall?

Reply:
As far as I can see, the old Fp serialization simply called `self.write(&mut bytes[..])?;` (from the `ToBytes` trait), which in turn calls `ToBytes` on the `.into_repr()` (which is a BigInteger).
So, it probably has the same pitfall right now. But it is an interesting point!

Reply:
We avoided it in the past because we require the output_buf to have length equal to the byte size of the field, and copied from the 64-bit aligned serialized struct:

https://github.com/scipr-lab/zexe/blob/603637946a8e88685df75a0d7b007751730bc286/algebra-core/src/serialize/mod.rs#L39

Reply:
Ok, good catch then. I’ll add this aspect to my serialization as well.